spark/dl/src/main/scala/com/intel/analytics/bigdl/transform/vision/image/ImageFeature.scala

```diff
@@ -200,12 +200,22 @@ class ImageFeature extends Serializable {
    */
   def copyTo[T: ClassTag](storage: Array[T], offset: Int, floatKey: String = ImageFeature.floats,
     toRGB: Boolean = true)(implicit ev: TensorNumeric[T]): Unit = {
+    val channel = getChannel()
     require(contains(floatKey), s"there should be ${floatKey} in ImageFeature")
     val data = floats(floatKey)
-    require(data.length >= getWidth() * getHeight() * 3,
+    require(data.length >= getWidth() * getHeight() * channel,
       s"float array length should be larger than 3 * ${getWidth()} * ${getHeight()}")
```

Comment:
float array length should be larger than 3 * ${getWidth()} * ${getHeight()} => float array length should be larger than  ${getChannel() ? }* ${getWidth()} * ${getHeight()} *